src/__tests__/index.js

```diff
@@ -105,7 +105,7 @@ const tests = {
   },
   'can handle objected with nested keys': {
     input: [
-      [{name: {first: 'baz'}}, {name: {first: 'bat'}}, {name: {first: 'foo'}}],
+      [{name: {first: 'baz'}}, {name: {first: 'bat'}}, {}],
```

Comment:
Could you actually add another item to the array? This test is also ensuring that we filter out the foo item.

Reply:
oh right gotcha.